drop-in/src/main/java/com/adyen/checkout/dropin/ui/DropInActivity.kt

```diff
@@ -164,8 +164,7 @@ class DropInActivity : AppCompatActivity(), DropInBottomSheetDialogFragment.Prot
 
         val bundle = savedInstanceState ?: intent.extras
 
-        val initializationSuccessful = assertBundleExists(bundle)
-        if (!initializationSuccessful) {
+        if (!DropInViewModel.assertBundleExists(bundle)) {
```

Comment:
Why is this a static call?

Reply:
The crash happens right when `DropInViewModel` is being initialised, so it would have to be a method that gets called before the view model is initialised.

I decided to put in `DropInViewModel.Companion` because we also have `putIntentExtras` there which is very related to `assertBundleExists`.

How do you suggest we improve this?

Reply:
This is in `onCreate`, right? If so, that means the ViewModel is already available. `assertBundleExists` doesn't need anything else from the ViewModel, so I think we can make it a normal, non-static method

Reply:
The view model is lazily initialised meaning even inside `onCreate` it's not created yet. And the crash happens on init because some `val`s inside it depend on the saved state handle having these keys present there.

Reply:
Hmm I see now